### PR TITLE
Send a confirmation email when the order is canceled or refunded

### PR DESCRIPTION
**Pull request by IKarbowiak**
Send a confirmation email when the order is canceled or refunded.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [x] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/order/actions.py [hunk 1]
```diff
@@ -83,6 +88,8 @@ def cancel_order(order: "Order", user: Optional["User"]):
     manager.order_cancelled(order)
     manager.order_updated(order)
 
+    send_order_canceled_confirmation(order, user)
```
Comment (maarcingebala) [inline saleor/order/actions.py:91]: Not a big issue but sometimes we say "cancelled" and sometimes "canceled" - both are correct but we have them mixed in entire codebase. Do you have any opinion on which one should we use @patrys?

File: saleor/order/emails.py [hunk 2]
```diff
@@ -88,7 +96,7 @@ def send_order_confirmation(order_pk, redirect_url, user_pk=None):
         order=email_data["context"]["order"],
         user=None,
         user_pk=user_pk,
-        email_type=events.OrderEventsEmails.ORDER,
+        email_type=events.OrderEventsEmails.ORDER_CONFIRMATION,
```
Comment (maarcingebala) [inline saleor/order/emails.py:99]: This is change is probably correct but we need to confirm with @dominik-zeglen if this doesn't break the events UI in the dashboard. I think we use these type fields for rendering there but I'm not sure.

Comment (maarcingebala) [reply saleor/order/emails.py:99]: If we used this `OrderEventsEmails.ORDER`, we will have to open a task for dashboard to change it to `ORDER_CONFIRMATION`.

File: saleor/order/actions.py (hunk 1 above)
Comment (patrys) [reply saleor/order/actions.py:91]: "Cancelled" is British, "canceled" is American.

Comment (maarcingebala) [reply saleor/order/actions.py:91]: Yeah I know, but do we have any preference on which to use?

Comment (patrys) [reply saleor/order/actions.py:91]: Looking at the code it seems American English is used almost everywhere (including the code itself).

File: saleor/order/emails.py (hunk 2 above)
Comment (dominik-zeglen) [reply saleor/order/emails.py:99]: Well, we do :)

Comment (maarcingebala) [reply saleor/order/emails.py:99]: Then please open a task to change it in the dashboard or if this is a simple replacement, we can fix it right away.

Comment (IKarbowiak) [reply saleor/order/emails.py:99]: I created a task.

**Pull request merged**